conditional.js

```diff
@@ -62,35 +75,93 @@ function addConditionals(loader) {
 				conditionModule = conditionModule.substr(1);
 			}
 
-			var handleConditionalBuild = function(m) {
+			var handleConditionalBuild = function() {};
+
+			//!steal-remove-start
+			handleConditionalBuild = function() {
+				var setBundlesPromise = Promise.resolve();
+
+				// make sure loader.bundle is an array
 				loader.bundle = typeof loader.bundle === "undefined" ?
 					[] : loader.bundle;
 
 				if (substitution) {
-					var cases = readMemberExpression("cases", m);
-
-					cases = isArray(cases) ? cases : [];
-
-					// if conditionModule define a `cases` property, use it
-					// to add bundles for each possible option that might be
-					// imported using string substitution.
-					for (var i = 0; i < cases.length; i += 1) {
-						var mod = cases[i];
-						var modName = name.replace(conditionalRegEx, mod);
-						var isBundle = loader.bundle.indexOf(modName) !== -1;
+					var glob = null;
+					var nameWithConditional = name;
+
+					// remove the conditional and the trailing slash
+					var nameWithoutConditional = name
+						.replace(conditionalRegEx, "")
+						.replace(/\/+$/, "");
+
+					setBundlesPromise = getGlob()
+					.then(function(nodeGlob) {
+						// in the browser we don't load the node modules
+						if (!nodeGlob) {
+							throw new Error("glob module not loaded");
+						}
 
-						if (modName && !isBundle) {
-							loader.bundle.push(modName);
+						// make glob available down the pipeline
+						glob = nodeGlob;
+
+						return normalize.call(loader, nameWithoutConditional,
+							parentName, parentAddress, pluginNormalize);
+					})
+					.then(function(normalized) {
+						return loader.locate({ name: normalized + "/*", metadata: {} });
+					})
+					.then(function(address) {
+						var path = address.replace("file:", "");
+						var parts = path.split("/");
+						var pattern = parts.pop();
+
+						return new Promise(function(resolve, reject) {
+							var options = {
+								cwd: parts.join("/"),
+								dot: true, nobrace: true, noglobstar: true,
+								noext: true, nodir: true
+							};
+
+							glob(pattern, options, function(err, files) {
+								if (err) { reject(err); }
+								resolve(files);
+							});
+						});
+					})
+					.then(function(variations) {
+						var promises = [];
+
+						for (var i = 0; i < variations.length; i += 1) {
+							var variation = variations[i].replace(".js", "");
+							var modName = nameWithConditional.replace(conditionalRegEx, variation);
+
+							var promise = loader.normalize.call(loader, modName,
+								parentName, parentAddress, pluginNormalize);
+
+							promises.push(promise.then(function(normalized) {
+								var isBundle = loader.bundle.indexOf(normalized) !== -1;
```

Comment:
Do we know that `loader.bundle` exists at this point? I don't think it will if the user hasn't set it up. Maybe you added a check somewhere else in this code?